packages/sql/src/credentials.d.ts

```diff
@@ -0,0 +1,5 @@
+export interface Credentials {
```

Comment:
Credentials should probably evolve to a Class in the future. The type + some methods (eg. creating a serverlURL from the original serverTemplate (with {user}) is done exactly in the same way in maps-api. 

Should not we consider a shared toolkit-core or utils?

Reply:
Yeah, I thought the same, but I don't how to solve this.

Keep in mind making it a separate package means we have to publish it on npm, and it would be the most pointless package ever (and there's lots of competition).

We could have a "private" package (so, a folder, not really a dependency) that we include, but I don't know if that is possible with TS.

However, we have to solve the credentials issue, because I think the natural way is to pass an object instead of a string, and just read the apiKey from there every time, so if it updates externally (i.e: OAuth), it will just work.